beluga/include/beluga/views/take_evenly.hpp

```diff
@@ -30,64 +30,144 @@ namespace beluga::views {
 
 namespace detail {
 
-/// Implementation detail for a take_evenly range adaptor object.
-struct take_evenly_fn {  // NOLINT(readability-identifier-naming)
-  /// Overload that implements the take_evenly algorithm.
-  /**
-   * \tparam Range A [sized range](https://en.cppreference.com/w/cpp/ranges/sized_range).
-   * \param range Source range from where to take elements.
-   * \param count Number of elements to take.
-   *
-   * If `count` or the range size are zero, it returns an empty range.
-   * If `count` is greater than the range size, it returns all the elements.
-   * The first and last elements of the range are always included.
-   */
+/// \cond detail
+
+template <class Range>
+struct take_evenly_view : public ranges::view_facade<take_evenly_view<Range>, ranges::finite> {
+ public:
+  static_assert(ranges::view_<Range>);
+  static_assert(ranges::forward_range<Range>);
+  static_assert(ranges::sized_range<Range>);
+
+  constexpr take_evenly_view() = default;
+
+  constexpr take_evenly_view(Range range, std::size_t count)
+      : range_{std::move(range)}, count_{count}, size_{ranges::size(range_)} {}
```

Comment:
@nahueespinosa hmm, range copies? Are we positively sure this will always be cheap?

Reply:
`Range` is constrained to be a `view` at this point. Views are supposed to be cheaply copyable or move-only.

Reply:
Ah, missed the `static_assert` on first read. Cool.